lib/charms/loki_k8s/v0/loki.py

```diff
@@ -45,13 +45,15 @@ def __init__(self, *args):
             ...
 
         def _provide_loki(self):
-            loki_server = LokiServer()
-
-            if loki_server.is_ready:
-                version = loki_server.version
+            try:
+                version = self._loki_server.version
                 self.loki_provider = LokiProvider(self, "logging")
-                self.unit.status = ActiveStatus()
+                self._stored.provider_ready = True
```

Comment:
Not sure why you need this if it is equivalent to `if self._loki_server.version` or similar.

Reply:
No, it's no necessary, it's statement from previous refactor logic.